### PR TITLE
Show invisible delimeters (within comments) when pretty printing.

### PR DESCRIPTION
**Pull request by nnethercote**
Because invisible syntax is really hard to work with!

r? @petrochenkov

### CODE REVIEW

**Comment by nnethercote** [comment]
My general approach here was straightforward: to make the obvious change in `rustc_ast_pretty` and then change all affected tests to accept the new output.

The only libs change here is change to the text describing the `None` variant in https://doc.rust-lang.org/proc_macro/enum.Delimiter.html.

File: src/test/ui/proc-macro/capture-macro-rules-invoke.stdout [hunk 1]
```diff
@@ -13,8 +14,10 @@ PRINT-BANG INPUT (DEBUG): TokenStream [
 ]
 PRINT-BANG INPUT (DISPLAY): 1 + 1, { "a" }, let a = 1;, String, my_name, 'a, my_val = 30,
 std::option::Option, pub(in some::path) , [a b c], -30
-PRINT-BANG RE-COLLECTED (DISPLAY): 1 + 1, { "a" }, let a = 1, String, my_name, 'a, my_val = 30,
-std :: option :: Option, pub(in some :: path), [a b c], - 30
+PRINT-BANG RE-COLLECTED (DISPLAY): /*⟪*/1 + 1/*⟫*/, /*⟪*/{ "a" }/*⟫*/, /*⟪*/let a = 1/*⟫*/,
```
Comment (petrochenkov) [inline src/test/ui/proc-macro/capture-macro-rules-invoke.stdout:17]:
```suggestion
PRINT-BANG RE-COLLECTED (DISPLAY): /*⟪*/ 1 + 1 /*⟫*/, /*⟪*/ { "a" } /*⟫*/, /*⟪*/ let a = 1 /*⟫*/,
```
Looks so noisy.
Maybe the should be given some more space and formatted as braces in `print_mac_common`.

Comment (petrochenkov) [reply src/test/ui/proc-macro/capture-macro-rules-invoke.stdout:17]: Also these are some math symbols, if I copypaste them into notepad++ they look like rectangles (unknown symbols).
```suggestion
PRINT-BANG RE-COLLECTED (DISPLAY): /*«*/ 1 + 1 /*»*/, /*«*/ { "a" } /*»*/, /*«*/ let a = 1 /*»*/,
```
This is how it looks if `LEFT/RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK`s are used instead (which are the oldest unicode bracket symbols).

**Comment by nnethercote** [comment]
I have made the suggested changes. I was surprised to find that, once the `print_mac_common` change was made, the `token_kind_to_string_ext` change is no longer necessary because the code in that function handling invisible delimiters is unreachable, at least for the test suite.

**Comment by nnethercote** [comment]
As a follow-on I am experimenting with returning `Delimiter::Invisible` tokens from `TokenCursor`. It's much nicer dealing with an error message like this:
```
error: expected expression, found `/*«*/`
```
than the previous, which was presumably:
```
error: expected expression, found ``
```

**Pull request merged**